core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -171,7 +168,7 @@ public InclusiveManifestEvaluator load(Integer specId) {
           new ScanEvent(table.toString(), snapshot.snapshotId(), rowFilter, schema()));
 
       Iterable<ManifestFile> matchingManifests = Iterables.filter(snapshot.manifests(),
-          manifest -> evalCache.getUnchecked(manifest.partitionSpecId()).eval(manifest));
+          manifest -> evalCache.get(manifest.partitionSpecId()).eval(manifest));
```

Comment:
What was wrong with the existing cache?

Reply:
Baseline has opinions about using Caffeine caches vs. Guava caches. There are many reasons to prefer Caffeine caches over Guava caches, largely around performance. The APIs are only marginally different - we see an example of an API difference here, where `cache.get` won't throw a checked `ExecutionException`; that API doesn't declare any checked exceptions being thrown at all.

We discuss some of the implications of Caffeine vs. Guava here: https://github.com/palantir/gradle-baseline/pull/317.

Reply:
Is Caffeine's API stable? How often does it change?

I'm all for a better implementation, but I want to make sure it's a good idea to add another dependency.

Reply:
Taking a look at Caffeine's pom it doesn't look like they pull in any dependencies apart from annotations: http://central.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/2.7.0/caffeine-2.7.0.pom

We can ask the authors about the API stability, but I can't imagine the APIs changing within a major version.

I'm not keen on changing the Baseline rules if we don't have to and if the way forward is relatively straightforward. If Baseline is encouraging us to do the right thing, we should do the right thing instead of compromising.

Reply:
Yeah, if this doesn't pull in transitive dependencies, then it sounds like a win. Guava's API here isn't reliable anyway, so it's probably better in the long run. Let's move to caffeine then!